conn.go

```diff
@@ -68,9 +76,14 @@ func Dial(network, addr string) (*Conn, error) {
 // DialTLS connects to the given address on the given network using tls.Dial
 // and then returns a new Conn for the connection.
 func DialTLS(network, addr string, config *tls.Config) (*Conn, error) {
-	c, err := tls.Dial(network, addr, config)
-	if err != nil {
-		return nil, NewError(ErrorNetwork, err)
+	dc, err := net.DialTimeout(network, addr, ldapTimeout)
+	if err != nil{
+		return nil, err
```

Comment:
Can we continue using `NewError` here for consistency?  Maybe just like it was before:

```
return nil, NewError(ErrorNetwork, err)
```